src/server/server_family.cc

```diff
@@ -1626,38 +1626,49 @@ void ServerFamily::FlushAll(CmdArgList args, ConnectionContext* cntx) {
   cntx->SendOk();
 }
 
+bool ServerFamily::DoAuth(ConnectionContext* cntx, std::string_view username,
+                          std::string_view password) const {
+  const auto* registry = ServerState::tlocal()->user_registry;
+  const bool is_authorized = registry->AuthUser(username, password);
+  if (is_authorized) {
+    cntx->authed_username = username;
+    auto cred = registry->GetCredentials(username);
+    cntx->acl_commands = cred.acl_commands;
+    cntx->keys = std::move(cred.keys);
+    cntx->ns = &namespaces.GetOrInsert(cred.ns);
+    cntx->authenticated = true;
+  }
+  return is_authorized;
+}
+
 void ServerFamily::Auth(CmdArgList args, ConnectionContext* cntx) {
   if (args.size() > 2) {
-    return cntx->SendError(kSyntaxErr);
+    cntx->SendError(kSyntaxErr);
+    return;
```

Comment:
Why change this? We do it in many places, it's more convenient and saves some verbosity..

Reply:
(same below in many places)

Reply:
well IMO I find `return <<void expression>>` confusing but I am happy to revert this if you find that it's confusing

Reply:
will change this soon

Reply:
Yes, please revert :) Thanks!

Reply:
done